sync15-adapter/examples/sync-pass.rs

```diff
@@ -498,16 +508,21 @@ fn prompt_record_id(e: &PasswordEngine, action: &str) -> Option<String> {
 
 fn main() -> Result<(), failure::Error> {
     env_logger::init();
-    let oauth_data = do_auth(false)?;
 
-    let scope = &oauth_data.keys["https://identity.mozilla.com/apps/oldsync"];
+    let cfg = Config::release()?;
+    let tokenserver_base_url = cfg.token_server_endpoint_url()?.join("../../")?.as_str().to_string();
+
+    let mut acct = do_auth(cfg)?;
+    let token = acct.get_oauth_token(&[SYNC_SCOPE])?.unwrap();
```

Comment:
I think this is fine for a quick example, however in a real app I would recommend following this flow instead:
1. get_oauth_token()
2. if returned None -> do oauth flow, otherwise continue
3. if use of token returns 401, goto 2